### ISSUE TITLE
Build fails due to memory. 

### ISSUE THREAD

**Issue by hollowgalaxy** [polyaxon/polyaxon#677]
### Describe the bug
Build fail if running Job/Experiment right after `polyaxon upload`.
```
Traceback (most recent call last): 
File "/polyaxon/polyaxon/dockerizer/dockerizer/initializer/init.py", line 97, in cmd nvidia_bin=nvidia_bin) 
File "/polyaxon/polyaxon/dockerizer/dockerizer/initializer/init.py", line 37, in init commit=commit) 
File "/polyaxon/polyaxon/dockerizer/dockerizer/initializer/download.py", line 18, in download extract_path=extract_path 
File "/usr/local/lib/python3.7/site-packages/polyaxon_client/api/project.py", line 210, in download_repo extract_path=extract_path) 
File "/usr/local/lib/python3.7/site-packages/polyaxon_client/transport/http_transport.py", line 211, in download extract_path=extract_path) 
File "/usr/local/lib/python3.7/site-packages/polyaxon_client/transport/http_transport.py", line 232, in untar tar.extractall(extract_path)
File "/usr/local/lib/python3.7/tarfile.py", line 2002, in extractall numeric_owner=numeric_owner)
File "/usr/local/lib/python3.7/tarfile.py", line 2044, in extract numeric_owner=numeric_owner) 
File "/usr/local/lib/python3.7/tarfile.py", line 2114, in _extract_member self.makefile(tarinfo, targetpath)
File "/usr/local/lib/python3.7/tarfile.py", line 2163, in makefile copyfileobj(source, target, tarinfo.size, ReadError, bufsize) 
File "/usr/local/lib/python3.7/tarfile.py", line 250, in copyfileobj dst.write(buf) OSError: [Errno 12] Cannot allocate memory
```
### To Reproduce

I'm using ubuntu 16.04. Script I use to run jobs usage: `script.sh job.yaml -u`, script.sh is below
How I get the error. I have some code I upload which contains a python file. In that file I have a  `print('hello world')`, If I run my script with the above command it breaks during build. Rerunning the command works. If I make a change, running a command fails to build. If I run it again, then it works.
TLDR: first job fails, rerunning job works.
*script.sh*

``` 
#!/bin/bash
set -e
cd $PWD
# get yaml
# determine if xp or job
YAML=$1
shift 1

if test -f "$YAML"; then
    echo "$YAML exist"
else
    echo "File $YAML does not exist."
    exit 1
fi

if [[ $# -ge 1 ]] && [[ $1 == '-u' ]]; then
  echo "Running: polyaxon upload"
  polyaxon upload
  shift 1
fi

TYPE=$(grep -ozP "version:\W*[0-9]*\W*kind:\W*\K([\w-_\\.@#\\$%&\\^]*).*" <"$YAML")
echo "Type of yaml given <$TYPE>."

OUTPUT=$(polyaxon run -f $YAML)
echo $OUTPUT
case $TYPE in 

  job)
      # TODO: get the number of job
      COUNT=$(grep -ozP "Job\W*\K(?:[0-9]*)" <<< $OUTPUT)
      polyaxon job -j $COUNT logs
      ;;
  experiment)
      COUNT=$(grep -ozP "Experiment\W*\K(?:[0-9]*)" <<< $OUTPUT)
      polyaxon experiment -xp $COUNT logs
      ;;
  group)
      COUNT=$(grep -ozP "Group\W*\K(?:[0-9]*)" <<< $OUTPUT)
      echo "No logs will be played. Group number is $COUNT."
      ;;
  *)
      echo "Error: $TYPE is not supported."
      exit 64
      ;;
  esac
```
build error.

### Expected behavior
Not crash.
### Environment
*job.yaml*
```
---
version: 1
kind: job
environment:
  resources:
    cpu:
      limits: 1
      requests: 1
build:
  dockerfile: polyaxon/Dockerfile
run:
  cmd:
    - ls polyaxon
    - python polyaxon/temp.py
    - sleep 10
```

**Comment by hollowgalaxy** [comment]
I also tried running
```
---
version: 1
kind: job
environment:
  resources:
    cpu:
      limits: 1
      requests: 1
    memory:
      requests: 5120
      limits: 5120

build:
  dockerfile: polyaxon/Dockerfile
run:
  cmd:
    - ls polyaxon
    - python polyaxon/temp.py
    - sleep 10
```

**Comment by hollowgalaxy** [comment]
Also the Dockerfile config file shown in the cli is not the most recent one. AKA is not being updated.

**Comment by hollowgalaxy** [comment]
![image](https://user-images.githubusercontent.com/52365584/74393991-69a98b80-4dc0-11ea-8223-79058714edca.png)
Zero changes were made to the Dockerfile. Only python code that is run with the cmd

**Comment by hollowgalaxy** [comment]
Bonus bug: I can't seem to update the Dockefile used for building the container.

**Comment by mouradmourafiq** [comment]
Maybe because you are not waiting for the upload to finish?
Not sure how the custom script works.

**Comment by hollowgalaxy** [comment]
I tried adding 30 second delay, which did not help. The custom script above was starting the run after uploading finished. Which makes me think that there is some internal magic going on that I would have to wait for. Not sure what I can do to get the status of ready to run.

**Comment by hollowgalaxy** [comment]
The script above does the following.
1. cd into place from which the script was run
2. get the yaml file (relative is fine because of step 1)
3. run the upload
4. start run
5. determine type of yaml (job, experiment, group)
6. if job or experiment, start logging.

**Comment by mouradmourafiq** [comment]
can you explain the logic behind your script? it seems like you can do this: `polyaxon run -f file.yaml -u -l` to get the same thing!

**Comment by mouradmourafiq** [comment]
Sorry, I think I missed your comment where you explain the logic. I think you only need step 1 and the run with the command I mentioned in the previous comment.

**Comment by hollowgalaxy** [comment]
I wasn't aware of `-i -u`. I'll try that and see if I still get the error.

**Comment by mouradmourafiq** [comment]
it's not `-i` it's `-l` like logs. Please do `polyaxon run --help` for more details.

**Comment by hollowgalaxy** [comment]
fixed

**Comment by hollowgalaxy** [comment]
Using `-l -u` now. Thanks for that tip. However still have error.

**Comment by mouradmourafiq** [comment]
are the jobs scheduled on the same node?

**Comment by hollowgalaxy** [comment]
Not seeing the error anymore. Perhaps I was trying to run while is still finishing the upload.

**Comment by hollowgalaxy** [comment]
@mouradmourafiq not sure. Wasn't able to recreate anymore.
